src/test/java/org/junit/platform/engine/test/TestEngineStub.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2015-2016 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v1.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.junit.platform.engine.test;
+
+import org.junit.platform.engine.EngineDiscoveryRequest;
+import org.junit.platform.engine.ExecutionRequest;
+import org.junit.platform.engine.TestDescriptor;
+import org.junit.platform.engine.TestEngine;
+import org.junit.platform.engine.UniqueId;
+
+/**
+ * @since 1.0
+ */
+public class TestEngineStub implements TestEngine {
+
+	private final String id;
+
+	public TestEngineStub() {
+		this(TestEngineStub.class.getSimpleName());
+	}
+
+	public TestEngineStub(String id) {
+		this.id = id;
+	}
+
+	@Override
+	public String getId() {
+		return this.id;
+	}
+
+	@Override
+	public TestDescriptor discover(EngineDiscoveryRequest discoveryRequest, UniqueId uniqueId) {
+		return new TestDescriptorStub(UniqueId.forEngine(getId()), getId());
+	}
+
+	@Override
+	public void execute(ExecutionRequest request) {
+	}
+
+}
```

Comment:
Have you considered checking the nullness of the parameters of all the methods in this class, using `Objects.requireNonNull` or Guava's `Preconditions.checkNotNull`?